File: frontend/src/services/donorService.js
// File: src/services/donorService.js
// Description: Contains functions for all Donor-related API calls.

import api from './api';

// --- Existing Functions ---
const getMyProfile = () => {
  return api.get('/donor/profile');
};

const bookAppointment = (appointmentData) => {
  return api.post('/donor/appointment', appointmentData);
};

const getMyDonationHistory = () => {
  return api.get('/donor/donations/history');
};

// --- NEW FUNCTIONS ---

// Function to get the logged-in donor's scheduled appointments
const getMyScheduledAppointments = () => {
  return api.get('/donor/appointments/scheduled');
};

// Function to cancel a specific appointment
const cancelAppointment = (appointmentId) => {
  return api.patch(`/donor/appointment/${appointmentId}/cancel`);
};


export default {
  getMyProfile,
  bookAppointment,
  getMyDonationHistory,
  getMyScheduledAppointments, // Export the new function
  cancelAppointment,          // Export the new function
};

File: backend spring/src/main/java/com/bloodbank/service/DonorService.java
// File: src/main/java/com/bloodbank/service/DonorService.java

package com.bloodbank.service;

import com.bloodbank.dto.request.AppointmentCreateRequest;
import com.bloodbank.dto.response.DonationRecordDto;
import com.bloodbank.entity.DonationAppointment;
import com.bloodbank.entity.DonorProfile;

import java.util.List;

public interface DonorService {

    // Existing methods
    DonorProfile getMyProfile();
    List<DonationRecordDto> getMyDonationHistory();
    DonationAppointment bookAppointment(AppointmentCreateRequest request);

    // --- NEW METHODS ---
    
    // Method to cancel an appointment
    DonationAppointment cancelAppointment(Long appointmentId);
    
    // Method to get a list of upcoming (scheduled) appointments
    List<DonationAppointment> getMyScheduledAppointments();
}